Stratis.Bitcoin/Utilities/LinqExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace System.Linq
+{
+    public static class LinqExtensions
+    {
+        public static long Median(this IEnumerable<long> source)
```

Comment:
> Maybe IReadOnlyNodesCollection can be renamed to INodesCollection read only is extrapolated from the exposed methods.

Too late, this is now an interface in NBitcoin/NStratis. I considered that someone may want to make a writable one too hence the ReadOnly